File: jan22/jan22.js
const readline = require('readline');

function guessTheNumber() {
    const lowerBound = 1;
    const upperBound = 100;
    const targetNumber = Math.floor(Math.random() * (upperBound - lowerBound + 1)) + lowerBound;

    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    console.log("Welcome to Guess the Number!");
    console.log(`Guess the number between ${lowerBound} and ${upperBound}`);

    let attempts = 0;

    rl.on('line', (userInput) => {
        const userGuess = parseInt(userInput);
        attempts++;

        if (userGuess === targetNumber) {
            console.log(`Congratulations! You guessed the correct number in ${attempts} attempts.`);
            rl.close();
        } else if (userGuess < targetNumber) {
            console.log("Too low! Try again.");
        } else {
            console.log("Too high! Try again.");
        }
    });
}

guessTheNumber();

File: jan22/jan22.java
import java.util.Scanner;

public class GuessTheNumber {
    public static void guessTheNumber() {
        int lowerBound = 1;
        int upperBound = 100;
        int targetNumber = (int) (Math.random() * (upperBound - lowerBound + 1) + lowerBound);

        System.out.println("Welcome to Guess the Number!");
        System.out.println("Guess the number between " + lowerBound + " and " + upperBound);

        Scanner scanner = new Scanner(System.in);
        int attempts = 0;

        while (true) {
            int userGuess = scanner.nextInt();
            attempts++;

            if (userGuess == targetNumber) {
                System.out.println("Congratulations! You guessed the correct number in " + attempts + " attempts.");
                break;
            } else if (userGuess < targetNumber) {
                System.out.println("Too low! Try again.");
            } else {
                System.out.println("Too high! Try again.");
            }
        }

        scanner.close();
    }

    public static void main(String[] args) {
        guessTheNumber();
    }
}